driver-core/src/test/unit/com/mongodb/MongoCredentialSpecification.groovy

```diff
@@ -280,4 +280,28 @@ class MongoCredentialSpecification extends Specification {
 
         !credentialOne.toString().contains(password)
     }
+
+    def 'testEqualsAndHashCode'() {
+        expect:
+        credential() == credential()
```

Comment:
By wrapping the creation in a closure `{}` it means each call to the closure will produce a different instance.  This is so we can test the equality properly and not shortcut out with the same instance.